line-bot-api-client/src/main/java/com/linecorp/bot/client/LineOAuthClientBuilder.java

```diff
@@ -0,0 +1,179 @@
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
+import com.fasterxml.jackson.module.paramnames.ParameterNamesModule;
+
+import lombok.NonNull;
+import lombok.Setter;
+import lombok.ToString;
+import lombok.experimental.Accessors;
+import lombok.experimental.PackagePrivate;
+import okhttp3.Interceptor;
+import okhttp3.OkHttpClient;
+import okhttp3.logging.HttpLoggingInterceptor;
+import okhttp3.logging.HttpLoggingInterceptor.Level;
+import retrofit2.Retrofit;
+import retrofit2.converter.jackson.JacksonConverterFactory;
+
+@ToString
+@Accessors(fluent = true)
+public class LineOAuthClientBuilder {
+    /**
+     * Creates a {@link LineOAuthClient}. Use {@link LineOAuthClient#builder()} instead.
+     */
+    @PackagePrivate
+    LineOAuthClientBuilder() {
+    }
+
+    /**
+     * API Endpoint.
+     *
+     * <p>Default value = {@value}.
+     */
+    @Setter
+    private String apiEndPoint = LineClientConstants.DEFAULT_API_END_POINT;
+
+    /**
+     * Connection timeout.
+     *
+     * <p>Default value = {@value LineClientConstants#DEFAULT_CONNECT_TIMEOUT_MILLIS}ms.
+     */
+    @Setter
+    private long connectTimeout = LineClientConstants.DEFAULT_CONNECT_TIMEOUT_MILLIS;
+
+    /**
+     * Connection timeout.
+     *
+     * <p>Default value = {@value LineClientConstants#DEFAULT_READ_TIMEOUT_MILLIS}ms.
+     */
+    @Setter
+    private long readTimeout = LineClientConstants.DEFAULT_READ_TIMEOUT_MILLIS;
+
+    /**
+     * Write timeout.
+     *
+     * <p>Default value = {@value LineClientConstants#DEFAULT_WRITE_TIMEOUT_MILLIS}ms.
+     */
+    @Setter
+    private long writeTimeout = LineClientConstants.DEFAULT_WRITE_TIMEOUT_MILLIS;
+
+    /**
+     * Custom {@link Retrofit.Builder} used internally.
+     *
+     * <p>If you want to use your own setting, specify {@link Retrofit.Builder} instance.
+     * Default builder is used in case of {@code null} (default).
+     *
+     * <p>To use this method, please add dependency to 'com.squareup.retrofit2:retrofit'.
+     *
+     * @see #createDefaultRetrofitBuilder()
+     */
+    @Setter
+    private Retrofit.Builder retrofitBuilder;
+
+    private OkHttpClient.Builder okHttpClientBuilder;
+
+    /**
+     * Custom interceptors.
+     *
+     * <p>You can add your own interceptors.
+     */
+    @Setter
+    private List<Interceptor> additionalInterceptors = new ArrayList<>();
+
+    /**
+     * Set customized OkHttpClient.Builder.
+     *
+     * <p>In case of you need your own customized {@link OkHttpClient},
+     * this builder allows specify {@link OkHttpClient.Builder} instance.
+     *
+     * <p>To use this method, please add dependency to 'com.squareup.retrofit2:retrofit'.
```

Comment:
[q]
Is this dependency necessary to use this method? I think this method doesn't depends on the package (retrofit).

Reply:
I just followed this comment. https://github.com/line/line-bot-sdk-java/blob/master/line-bot-api-client/src/main/java/com/linecorp/bot/client/LineMessagingClientBuilder.java#L103
But hm, I now don't think need to keep this. Removed from OAuthClientBuilder. And will remove from LineMessagingClientBuilder in near future.